app.py

```diff
@@ -153,6 +153,357 @@ def sent_escape(sent):
 	senttok = sent.split()
 	return " ".join(senttok_escape(senttok))
 
+class ActivedopTree:
```

Comment:
Great to have this functionality bundled under a class. As it's getting large I would put it in a new file (activedoptree.py).